src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.OSX.cs

```diff
@@ -341,72 +343,97 @@ private void WatchForFileSystemEventsThreadStart(object arg)
                             relativePath = eventPaths[i].Remove(0, _fullDirectory.Length);
                         }
 
-                        // Check if this is a rename
-                        if (IsFlagSet(eventFlags[i], Interop.EventStream.FSEventStreamEventFlags.kFSEventStreamEventFlagItemRenamed))
+                        var changedFlags = FSEventStreamEventFlags.kFSEventStreamEventFlagItemInodeMetaMod |
```

Comment:
Can we move some of this stuff up earlier so that we only do the string comparison and potentially create the string via Remove if the right conditions are present with changedFlags?